modules/apps/map/map-common/src/main/resources/META-INF/resources/js/MapBase.js

```diff
@@ -62,15 +60,40 @@ const pendingCallbacks = {};
  * @abstract
  * @review
  */
-class MapBase extends State {
+class MapBase extends EventEmitter {
+	get position() {
+		return this._STATE_.position;
+	}
+
+	set position(position) {
+		this.emit('positionChange', {
+			newVal: {
+				address: position.address,
+				location: position.location,
+			},
+		});
+
+		this._STATE_.position = position;
+	}
 
 	/**
 	 * MapBase constructor
 	 * @param  {Array} args List of arguments to be sent to State constructor
 	 * @review
 	 */
-	constructor(...args) {
-		super(...args);
+	constructor(args = {}) {
+		super(args);
+
+		const {controls, geolocation, position} = args;
+
+		this._STATE_ = {
+			position,
+		};
```

Comment:
I don't think you need this here since you are setting `this.position = position` below. You may need to initialize `this._STATE_ = {}` here though